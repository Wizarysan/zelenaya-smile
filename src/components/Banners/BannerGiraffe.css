@media screen and (max-width: 767px) {
  .mobile-hide {
    display: none; }
  .mobile-show {
    display: initial; } }

@media screen and (min-width: 768px) {
  .mobile-hide {
    display: initial; }
  .mobile-show {
    display: none; } }

.animation__photo-enter {
  filter: blur(2px) grayscale(100%); }

.animation__photo-enter-active.animation__photo-enter {
  filter: blur(0px) grayscale(0%);
  transition: all 200ms ease-in; }

.animation__photo-leave {
  opacity: 1; }
  .animation__photo-leave-active.animation__photo-leave {
    opacity: 0.01;
    transition: opacity 200ms ease-in; }

@keyframes bump {
  0% {
    transform: scale(1); }
  50% {
    transform: scale(1.6); }
  100% {
    transform: scale(1); } }

.animation__bump {
  animation: bump .5s ease-out; }

@media (max-width: 767px) {
  .giraffe {
    display: none; } }

.giraffe__wrapper {
  position: relative;
  max-width: 1400px;
  margin: 0 auto;
  overflow: hidden; }

.giraffe__title {
  display: inline-block;
  font-size: 25px;
  padding: 70px 0 70px 30px; }
  .giraffe__title-big {
    transition: all .3s ease-out;
    font-family: 'unisans_hc', Arial;
    font-size: 61px;
    line-height: 0.8;
    color: #73bd4c;
    margin-left: -5px; }

.giraffe__body {
  display: inline-block;
  vertical-align: top; }

.giraffe__options {
  display: inline-block;
  vertical-align: top;
  text-align: center;
  padding-top: 65px;
  font-size: 16px;
  margin-left: 15px; }
  .giraffe__options-divider {
    display: none; }

.giraffe__option {
  display: inline-block;
  margin-left: 30px;
  margin-bottom: 15px;
  line-height: 1.2; }
  .giraffe__option-img {
    margin-bottom: 10px; }

.giraffe__head {
  position: absolute;
  right: 7%;
  top: -30px;
  transition: all .6s ease-out; }

.giraffe:hover .giraffe__head {
  top: -10px;
  transform: rotate(5deg); }

.giraffe:hover .giraffe__title-big {
  color: #88cd40; }

.giraffe:hover .giraffe__whole {
  transform: rotate(5deg); }

@media screen and (max-width: 1280px) {
  .giraffe__body {
    margin-left: -52px; }
  .giraffe__options {
    padding-top: 65px;
    margin-left: -40px; } }

@media screen and (max-width: 1155px) {
  .giraffe__options {
    padding-top: 15px;
    font-size: 15px;
    margin-left: 50px; }
    .giraffe__options-divider {
      display: block; }
  .giraffe__option {
    line-height: 1.0; }
    .giraffe__option-img {
      width: 57px; } }

@media screen and (max-width: 1080px) {
  .giraffe__options {
    margin-left: 30px; } }

@media screen and (max-width: 1020px) {
  .giraffe__options {
    margin-left: -20px; } }

@media screen and (max-width: 930px) {
  .giraffe__body {
    display: none; }
  .giraffe__head {
    position: relative;
    right: 0;
    margin-left: 5px;
    vertical-align: top; }
  .giraffe__options {
    margin-left: 15px; } }

.giraffe-v .giraffe {
  background: #f0f0f0; }
  .giraffe-v .giraffe__wrapper {
    height: 100vh; }
  .giraffe-v .giraffe__title {
    padding: 30px 0 0 0;
    display: block;
    text-align: center;
    font-size: 1.7vw; }
    .giraffe-v .giraffe__title-big {
      font-size: 2.4vw;
      line-height: 1.1; }
  .giraffe-v .giraffe__whole {
    padding: 30px 14% 0 0;
    position: absolute;
    left: -40px;
    bottom: 0;
    transition: all 1s; }
    .giraffe-v .giraffe__whole-img {
      max-height: 73vh; }
  .giraffe-v .giraffe__options {
    font-size: 13px;
    position: absolute;
    right: 10%;
    bottom: 100px;
    margin-left: 0; }
    .giraffe-v .giraffe__options-divider {
      display: block; }
  .giraffe-v .giraffe__option {
    margin-left: 18px; }
    .giraffe-v .giraffe__option-img {
      width: 50px; }

@media screen and (max-width: 1430px) {
  .giraffe-v .giraffe__whole {
    left: -70px; } }

@media screen and (max-width: 1315px) {
  .giraffe-v .giraffe__option {
    margin-left: 10px; }
  .giraffe-v .giraffe__options {
    font-size: 12px; } }

@media screen and (max-width: 1200px) {
  .giraffe-v .giraffe__option-img {
    width: 45px; }
  .giraffe-v .giraffe__options {
    right: 7%; }
  .giraffe-v .giraffe__title {
    padding: 40px 0 0 0; } }

@media screen and (max-width: 1100px) {
  .giraffe-v .giraffe__whole {
    left: -90px; } }

@media screen and (max-width: 1010px) {
  .giraffe-v .giraffe__whole {
    left: -90px; } }

@media screen and (max-width: 954px) {
  .giraffe-v .giraffe__whole {
    bottom: -52%;
    left: -80px; }
  .giraffe-v .giraffe__title {
    padding: 15px 0 0 0; }
  .giraffe-v .giraffe__options {
    font-size: 13px;
    bottom: 140px; }
  .giraffe-v .giraffe__option {
    padding: 0 5px; }
    .giraffe-v .giraffe__option-img {
      width: 40px; }
    .giraffe-v .giraffe__option-br {
      display: none; } }
