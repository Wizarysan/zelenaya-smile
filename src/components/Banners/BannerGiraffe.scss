@import './../../styles/shared.scss';

.giraffe {
  @include breakpoint(xs) {
    display: none;
  }
  &__wrapper {
    position: relative;
    max-width: 1400px;
    margin: 0 auto;
    overflow: hidden;
  }
  &__title {
    display: inline-block;
    font-size: 25px;
    padding: 70px 0 70px 30px;
    &-big {
      transition: all .3s ease-out;
      font-family: 'unisans_hc', Arial;
      font-size:61px;
      line-height: 0.8;
      color: $br-green;
      margin-left: -5px;
    }
  }
  &__body {
    display: inline-block;
    vertical-align: top;
  }
  &__options {
    display: inline-block;
    vertical-align: top;
    text-align: center;
    padding-top: 65px;
    font-size: 16px;
    margin-left: 15px;
    &-divider {
      display: none;
    }
  }
  &__option {
    display: inline-block;
    margin-left: 30px;
    margin-bottom: 15px;
    line-height: 1.2;
    &-img {
      margin-bottom: 10px;
    }
  }
  &__head {
    position: absolute;
    right: 7%;
    top: -30px;
    transition: all .6s ease-out;
  }
  &:hover {
    .giraffe__head {
      top:-10px;
      transform: rotate(5deg);
    }
    .giraffe__title-big {
      color: #88cd40;
    }
    .giraffe__whole {
      transform: rotate(5deg);
    }
  }
}

@media screen and (max-width: 1280px) {
  .giraffe__body {
    margin-left: -52px;
  }
  .giraffe__options {
    padding-top: 65px;
    margin-left: -40px;
  }
}

@media screen and (max-width: 1155px) {
  .giraffe__options {
    padding-top: 15px;
    font-size: 15px;
    margin-left: 50px;
    &-divider {
      display: block;
    }
  }
  .giraffe__option {
    line-height: 1.0;
    &-img {
      width: 57px;
    }
  }
}

@media screen and (max-width: 1080px) {
  .giraffe__options {
    margin-left: 30px;
  }
}

@media screen and (max-width: 1020px) {
  .giraffe__options {
    margin-left: -20px;
  }
}

@media screen and (max-width: 930px) {
  .giraffe__body {
    display: none;
  }
  .giraffe__head {
    position: relative;
    right:0;
    margin-left:5px;
    vertical-align: top;
  }
  .giraffe__options {
    margin-left: 15px;
  }
}

.giraffe-v {
  .giraffe {
    background: #f0f0f0;
    &__wrapper {
      height: 100vh;
    }
    &__title {
      padding:30px 0 0 0;
      display: block;
      text-align: center;
      font-size: 1.7vw;
      &-big {
        font-size: 2.4vw;
        line-height: 1.1;
      }
    }
    &__whole{
      padding: 30px 14% 0 0;
      position:absolute;
      left: -40px;
      bottom: 0;
      transition: all 1s;
      &-img {
        max-height: 73vh;
      }
    }
    &__options {
      font-size: 13px;
      position: absolute;
      right: 10%;
      bottom: 100px;
      margin-left: 0;
      &-divider {
        display: block;
      }
    }
    &__option {
      margin-left: 18px;
      &-img {
        width: 50px;
      }
    }
  }
}

@media screen and (max-width: 1430px) {
  .giraffe-v {
    .giraffe__whole {
      left: -70px;
    }
  }
}
@media screen and (max-width: 1315px) {
  .giraffe-v {
    .giraffe__option {
      margin-left: 10px;
    }
    .giraffe__options {
      font-size: 12px;
    }
  }
}
@media screen and (max-width: 1200px) {
  .giraffe-v {
    .giraffe__option {
      &-img {
        width: 45px;
      }
    }
    .giraffe__options {
      right:7%;
    }
    .giraffe__title {
      padding:40px 0 0 0;
    }
  }
}
@media screen and (max-width: 1100px) {
  .giraffe-v {
    .giraffe__whole {
      left: -90px;
    }
  }
}
@media screen and (max-width: 1010px) {
  .giraffe-v {
    .giraffe__whole {
      left: -90px;
    }
  }
}
@media screen and (max-width: 954px) {
  .giraffe-v {
    .giraffe__whole {
      bottom: -52%;
      left: -80px;
    }
    .giraffe__title{
      padding: 15px 0 0 0;
    }
    .giraffe__options {
      font-size: 13px;
      bottom: 140px;
    }
    .giraffe__option {
      padding: 0 5px;
      &-img {
        width: 40px;
      }
      &-br {
        display:none;
      }
    }
  }
}
