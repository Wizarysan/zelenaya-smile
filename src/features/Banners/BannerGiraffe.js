import React from 'react';

//TODO выделить пути features в отдельный файл с константами

import giraffeBody from './../../assets/images/banners/giraffe-body.png';
import giraffeHead from './../../assets/images/banners/giraffe-head.png';
import giraffeWeb from './../../assets/images/banners/giraffe-web.png';
import giraffeTv from './../../assets/images/banners/giraffe-tv.png';
import giraffeMediac from './../../assets/images/banners/giraffe-mediac.png';
import giraffePlay from './../../assets/images/banners/giraffe-play.png';
import './BannerGiraffe.css';

//TODO выбор города для линка в баннере

const BannerGiraffe = (props) => {
  return(
    <div className="giraffe back-greygrad">
      <a href={`http://${props.cityGW}.zelenaya.net/mixer/`} >
        <div className="giraffe__wrapper">
          <div className="giraffe__title">
            <span>интернет и тв</span>
            <div className="giraffe__title-big">ОГРОМНЫХ</div>
            <span>возможностей</span>
          </div>
          <div className="giraffe__body">
            <img src={giraffeBody} className="giraffe__body-img"/>
          </div>
          <div className="giraffe__options">
            <div className="giraffe__option">
              <img src={giraffeWeb} className="giraffe__option-img"/>
              <div>Интернет<br/> тарифы до<br/> 100 Мбит/с</div>
            </div>
            <div className="giraffe__option">
              <img src={giraffeTv} className="giraffe__option-img"/>
              <div>Цифровое ТВ<br/>более 250<br/> каналов</div>
            </div>
            <br className="giraffe__options-divider"/>
            <div className="giraffe__option">
              <img src={giraffePlay} className="giraffe__option-img"/>
              <div>Бесплатное<br/> управление<br/> ТВ-эфиром</div>
            </div>
            <div className="giraffe__option">
              <img src={giraffeMediac} className="giraffe__option-img"/>
              <div>Бесплатный<br/> медиацентр<br/>в комплекте</div>
            </div>
          </div>
          <img src={giraffeHead} className="giraffe__head"/>
        </div>
      </a>
    </div>
  )
}

export default BannerGiraffe;
