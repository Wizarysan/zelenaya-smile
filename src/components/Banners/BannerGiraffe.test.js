import React from 'react';
import BannerGiraffe from './BannerGiraffe.js';
import renderer from 'react-test-renderer';

it('renders correctly with test city', () => {
  const tree = renderer
    .create(<BannerGiraffe cityGW="testCity"/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});