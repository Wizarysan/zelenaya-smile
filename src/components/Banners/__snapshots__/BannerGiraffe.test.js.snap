exports[`test renders correctly with test city 1`] = `
<div
  className="giraffe back-greygrad">
  <a
    href="http://testCity.zelenaya.net/mixer/">
    <div
      className="giraffe__wrapper">
      <div
        className="giraffe__title">
        <span>
          интернет и тв
        </span>
        <div
          className="giraffe__title-big">
          ОГРОМНЫХ
        </div>
        <span>
          возможностей
        </span>
      </div>
      <div
        className="giraffe__body">
        <img
          className="giraffe__body-img"
          src="giraffe-body.png" />
      </div>
      <div
        className="giraffe__options">
        <div
          className="giraffe__option">
          <img
            className="giraffe__option-img"
            src="giraffe-web.png" />
          <div>
            Интернет
            <br />
             тарифы до
            <br />
             100 Мбит/с
          </div>
        </div>
        <div
          className="giraffe__option">
          <img
            className="giraffe__option-img"
            src="giraffe-tv.png" />
          <div>
            Цифровое ТВ
            <br />
            более 250
            <br />
             каналов
          </div>
        </div>
        <br
          className="giraffe__options-divider" />
        <div
          className="giraffe__option">
          <img
            className="giraffe__option-img"
            src="giraffe-play.png" />
          <div>
            Бесплатное
            <br />
             управление
            <br />
             ТВ-эфиром
          </div>
        </div>
        <div
          className="giraffe__option">
          <img
            className="giraffe__option-img"
            src="giraffe-mediac.png" />
          <div>
            Бесплатный
            <br />
             медиацентр
            <br />
            в комплекте
          </div>
        </div>
      </div>
      <img
        className="giraffe__head"
        src="giraffe-head.png" />
    </div>
  </a>
</div>
`;
